Directory.Build.props

```diff
@@ -332,6 +331,11 @@
 
     <!-- this property is used by the SDK to pull in mono-based runtime packs -->
     <UseMonoRuntime Condition="'$(UseMonoRuntime)' == '' and '$(RuntimeFlavor)' == 'Mono'">true</UseMonoRuntime>
+    
+    <!-- For enabling the use of XUnitLogChecker in coreclr and libraries test runs. -->
+    <IsXUnitLogCheckerSupported Condition="'$(IsXUnitLogCheckerSupported)' == ''">false</IsXUnitLogCheckerSupported>
+    <IsXUnitLogCheckerSupported Condition="'$(RuntimeFlavor)' == 'CoreCLR' and '$(TestNativeAot)' != 'true' and '$(TestRunNamePrefixSuffix)' != 'NativeAOT_Release' and '$(TargetOS)' != 'browser' and '$(TargetOS)' != 'wasi' and '$(TargetOS)' != 'ios' and '$(TargetOS)' != 'iossimulator' and '$(TargetOS)' != 'tvos' and '$(TargetOS)' != 'tvossimulator' and '$(TargetOS)' != 'maccatalyst' and '$(TargetOS)' != 'android'">true</IsXUnitLogCheckerSupported>
```

Comment:
Why do you need to condition on `TestRunNamePrefixSuffix`? NativeAOT is already excluded via the `TestNativeAot` property, isn't it?

Reply:
Do you need both in that case? Also, runtime flavor should remove a lot of the TargetOS ones.

Reply:
I noticed some NativeAot runs were not getting ignored unless I added both conditions. They are temporary anyway. I opened an issue so the NativeAOT owners can get it enabled properly.

Reply:
Having to condition on `'$(TestRunNamePrefixSuffix)' != 'NativeAOT_Release'` is extremely suspicious. We only ever run native AOT testing with `TestNativeAot`. It's not native AOT testing if that property is not true. This will likely produce __obscure__ failure modes if we add new test pipelines or rename existing. It will be expensive to root cause for whoever isn't already aware of this hack. Cc @dotnet/ilc-contrib for awareness.

> They are temporary anyway. I opened an issue so the NativeAOT owners can get it enabled properly.

Unless you already talked to some "NativeAOT owners" and scheduled their time to fix this, this temporary workaround is likely to be in the product for a long time, like most temporary workarounds. I'm very much against this condition being checked in as it is, please investigate this. One thing off top of my head is that this might need to check for `RunNativeAotTestApps` instead, which gates special kinds of library tests that was added in #76109.

Reply:
@MichalStrehovsky Since I had to revert the change from main due to NativeAOT tests being broken in outerloop, I can remove the TestRunNamePrefixSuffix property from the condition in the new PR.
I also opened https://github.com/dotnet/runtime/issues/94722 to track enabling them in the future.

Reply:
Thanks! The issue might have some overlap with #94270.

Reply:
Feel free to mark mine as duplicate of yours if they're tracking the same.